app/waf/src/models/profiles/http_compress.model.ts

```diff
@@ -0,0 +1,376 @@
+/**
+ * Copyright 2019 F5 Networks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {model, property} from '@loopback/repository';
+import {CommonEntity} from '../common.model';
+
+@model()
+export class ProfileHTTPCompression extends CommonEntity {
+  @property({
+    type: 'boolean',
+    required: false,
+    default: false,
+    schema: {
+      create: true,
+      update: true,
+      response: true,
+      example: true,
+      openapi: {
+        enum: [true, false],
+      },
+    },
+    as3: {},
+  })
+  allowHTTP10?: boolean;
+
+  @property({
+    type: 'number',
+    required: false,
+    default: 4096,
+    schema: {
+      create: true,
+      update: true,
+      response: true,
+      example: 4096,
+      openapi: {
+        type: 'integer',
+        minimum: 256,
+        maximum: 32768,
+      },
+    },
+    as3: {},
+  })
+  bufferSize?: number;
+
+  @property({
+    type: 'array',
+    itemType: 'string',
+    required: false,
+    default: [],
+    schema: {
+      create: true,
+      update: true,
+      response: true,
+      example: ['application/json', 'text/xml'],
+      openapi: {
+        items: {
+          type: 'string',
+        },
+      },
+    },
+    as3: {},
+  })
+  contentTypeExcludes?: string;
+
+  @property({
+    type: 'array',
+    itemType: 'string',
+    required: false,
+    default: ['text/', 'application/(xml|x-javascript)'],
+    schema: {
+      create: true,
+      update: true,
+      response: true,
+      example: ['application/json', 'text/xml'],
+      openapi: {
+        items: {
+          type: 'string',
+        },
+      },
+    },
+    as3: {},
+  })
+  contentTypeIncludes?: string;
+
+  @property({
+    type: 'boolean',
+    required: false,
+    default: true,
+    schema: {
+      create: true,
+      update: true,
+      response: true,
+      example: true,
+      openapi: {
+        enum: [true, false],
+      },
+    },
+    as3: {},
+  })
+  cpuSaver?: boolean;
```

Comment:
It should always have a value, so it does not need this '?'.

Reply:
removed.